packages/shared/src/components/auth/EmailCodeVerification.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { ReactElement, useContext, useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { Button } from '../buttons/Button';
+import { TextField } from '../fields/TextField';
+import { AuthFormProps } from './common';
+import { AuthFlow } from '../../lib/kratos';
+import useAccountEmailFlow from '../../hooks/useAccountEmailFlow';
+import AuthForm from './AuthForm';
+import { KeyIcon, MailIcon, VIcon } from '../icons';
+import { AuthEventNames } from '../../lib/auth';
+import AnalyticsContext from '../../contexts/AnalyticsContext';
+import Alert, { AlertParagraph, AlertType } from '../widgets/Alert';
+
+interface EmailCodeVerificationProps extends AuthFormProps {
+  code?: string;
+  email: string;
+  flowId: string;
+  onSubmit?: () => void;
+  className?: string;
+}
+function EmailCodeVerification({
+  code: codeProp,
+  email: emailProp,
+  flowId,
+  onSubmit,
+  className,
+}: EmailCodeVerificationProps): ReactElement {
+  const { trackEvent } = useContext(AnalyticsContext);
+  const [hint, setHint] = useState('');
+  const [alert, setAlert] = useState({ firstAlert: true, alert: false });
+  const [code, setCode] = useState(codeProp);
+  const [email, setEmail] = useState(emailProp);
+  const { sendEmail, verifyCode, resendTimer, isLoading, autoResend } =
+    useAccountEmailFlow({
+      flow: AuthFlow.Verification,
+      flowId,
+      onError: setHint,
+      onVerifyCodeSuccess: () => {
+        onSubmit();
+      },
+    });
+
+  useEffect(() => {
+    if (autoResend && !alert.alert && alert.firstAlert === true) {
```

Comment:
If i understood this right, firstAlert seems to be to keep this running only once. I think it's better if alert can have states maybe:

Maybe is better to leave it as null or undefined and look specifically for undefined type in the first try, rather then a boolean value.

Reply:
Ah yes, totally get your message, I initially wanted to solve it with one state, but because everything is so intertwined from propdrilling this became super messy and introduced frontend bugs.
I'll see if I can quickly make it a bit easier.